lib/signal_tower.ex

```diff
@@ -45,4 +40,16 @@ defmodule SignalTower do
     Logger.info("SignalTower started")
     ret
   end
+
+  @impl Application
+  def stop(_) do
+    DynamicSupervisor.which_children(Room.Supervisor)
+    |> Enum.map(fn
+      {:undefined, pid, :worker, _} when is_pid(pid) ->
+        send(pid, :shutdown)
+
+      _ ->
+        :ok
+    end)
+  end
```

Comment:
Is the signal handler?

Reply:
+ this

Reply:
... Is this the signal handler

Reply:
What do you mean by signal handler? stop(state) is a callback of the Application behaviour ("interface") and it gets called when the application is asked to shut down - is that what you mean?

Reply:
Thats what i mean. I was refering to signal handlers used in c++ programms.

Reply:
Is the 'asked to shutdown' compareable to unix signal e.g.: sigterm?

Reply:
Exactly. "By default, a SIGTERM from the operating system will automatically translate to System.stop/0." https://hexdocs.pm/elixir/Application.html

Reply:
Thx a lot for clarification!